### PR TITLE
[env/ohbm] set venuePage HTML document title basing off PLATFORM_BRAND_NAME

### PR DESCRIPTION
**Pull request by yarikoptic**
This greatly improves UX (User eXperience) by not only making page/tab
show the location, but also making browser history useful.  Without this
change you get the exciting "Sparkle" as the title (hardcoded in index.html)
for every page.

Joint work with @margulies where I was "git grep"ing and he was editing and trying it out.

1st commit is generic and could be applied (cherrypicked) to the main code base.  2nd commit is OHBM vendoring. Let me know if there is another/better way to vendor it.

### CODE REVIEW

**Comment by 0xdevalias** [comment]
[sparkle-internal-note] See https://github.com/sparkletown/internal-sparkle-issues/issues/19 for our backlog issue related to implementing this feature + notes on how we intended to do so.

File: src/pages/VenuePage/VenuePage.tsx
Comment (0xdevalias) [inline src/pages/VenuePage/VenuePage.tsx:120]: As we already include `react-use` in our dependencies, we could use `useTitle` here:

- https://github.com/streamich/react-use/blob/master/docs/useTitle.md

**Review by 0xdevalias — CHANGES_REQUESTED** [review]
Hey @yarikoptic,

Thanks for the PR :)

I've left a small improvement comment below, it would be cool if you could make that change.

I'll need to check with the team (@mike-lvov FYI) what our specific plan is for merging these `env/ohbm` specific customisations are (so that it's still easy for us to merge in the 'main branch' features), but I expect this should be fine once those changes are made.

Thanks again!

**Comment by yarikoptic** [comment]
> [sparkle-internal-note] See [sparkletown/internal-sparkle-issues#19](https://github.com/sparkletown/internal-sparkle-issues/issues/19) for our backlog issue related to implementing this feature + notes on how we intended to do so.

must be private since 404s for me, so I just proceeded:  Thanks for the guidance. I rewrote using `setTitle` (also sorted imports nearby while at it) -- seems to work.

> 'll need to check with the team (@mike-lvov FYI) what our specific plan is for merging these `env/ohbm` specific customisations are

for now, I just reworded vendoring commit to have "ohbm: " prefix to annotate for its specificity to possibly ease future cherry-picking etc.

File: src/pages/VenuePage/VenuePage.tsx
Comment (yarikoptic) [reply src/pages/VenuePage/VenuePage.tsx]: use that now

**Comment by 0xdevalias** [comment]
> must be private since 404s for me

@yarikoptic Yup, that's why I marked it as `[sparkle-internal-note]`

---

> Thanks for the guidance. I rewrote using `setTitle` (also sorted imports nearby while at it) -- seems to work.

@yarikoptic Cool! Sounds good :)

---

> for now, I just reworded vendoring commit to have "ohbm: " prefix to annotate for its specificity to possibly ease future cherry-picking etc.

@yarikoptic Sounds good :)

**Comment by 0xdevalias** [comment]
@yarikoptic Out of curiosity, how comfortable are you with git? If I asked you to rebase this branch off `staging` rather than off `env/ohbm` so we can retarget this to `staging`; would you know how to do that?

If so, I think the ideal situation here would be for you to make a new PR against `env/ohbm` with just the OHBM specific tweaks, and then we can have the main `useTitle` stuff merged into the core codebase.

For merging into the core codebase, I would change `useTitle(PLAYA_VENUE_NAME + ": " + venueName);` to:

```typescript
useTitle(`${PLATFORM_BRAND_NAME} - ${venueName}`);
```

Add `PLATFORM_BRAND_NAME` in `src/settings` (somewhere near `SPARKLE_HOMEPAGE_URL` probably)

Then in the `env/ohbm` specific PR, instead of using `PLAYA_VENUE_NAME`, just change `PLATFORM_BRAND_NAME`.

**Comment by yarikoptic** [comment]
Will do. git is my friend

**Comment by yarikoptic** [comment]
done : https://github.com/sparkletown/sparkle/pull/1389 . I kept this PR with both commits for now.  `git` would be fine having the same commit in both, but if that PR gets into staging and env/ohbm gets it merged before this PR, I can just drop it, or drop it right away if you insist.

**Comment by 0xdevalias** [comment]
> I kept this PR with both commits for now. `git` would be fine having the same commit in both, but if that PR gets into staging and env/ohbm gets it merged before this PR, I can just drop it, or drop it right away if you insist.

@yarikoptic It's fine to leave it here for now. Though I need to discuss our plans/process for merging the environment specific tweaks before I will merge this one, which likely won't be till early next week (midnight Friday night here in Australia at the moment)

**Comment by yarikoptic** [comment]
FTR: I guess I will wait for current `staging` to be merged into `env/ohbm` and then will drop the first commit which is already in `staging`.

**Comment by yarikoptic** [comment]
IMHO this PR is ready (contains only 1 tiny commit with vendoring setting) @0xdevalias

**Comment by yarikoptic** [comment]
@0xdevalias could you change the review from "Changes requested" to "Approved" or even merge it? ;-)

**Comment by 0xdevalias** [comment]
> could you change the review from "Changes requested" to "Approved" or even merge it? ;-)

@yarikoptic I will get to reviewing + merging this alongside all of the other reviews and merges that need to be done. I haven't approved and merged it yet because I am still needing to chat with @mike-lvov about our plans for merging the custom tweaks vs the main codebranch in a way that ensures we can efficiently deliver the main updates to the `env/ohbm` environment.

It's on my list, I will get to it when I have the resources to do so.

**Review by 0xdevalias — APPROVED** [review]
Discussed with @mike-lvov, we're comfortable shipping the environment specific tweaks directly to the `env/ohbm` branch now that the big work with the schedule/performance refactor have landed.

LGTM! :shipit:

**Pull request merged**
